### PR TITLE
feat: initialize admission webhook

### PR DESCRIPTION
**Pull request by daviderli614**
related issue: https://github.com/GreptimeTeam/greptimedb-operator/issues/153
related docs: https://kubebuilder.io/cronjob-tutorial/webhook-implementation

## Summary by CodeRabbit

- **New Features**
	- Introduced webhook configurations for `GreptimeDBCluster` and `GreptimeDBStandalone` resources, enabling validation.
	- Added webhooks for handling create, update, and delete operations for both resources.

- **Bug Fixes**
	- Updated resource path for `GreptimeDBStandalone` for consistency.

- **Documentation**
	- Enhanced deepcopy methods for various data structures to improve robustness.

### CODE REVIEW

File: PROJECT [hunk 1]
```diff
@@ -10,14 +14,20 @@ resources:
   controller: true
   domain: greptime.io
   kind: GreptimeDBCluster
-  path: github.com/GreptimeTeam/greptimedb-operator/apis/v1alpha1
+  path: github.com/GreptimeTeam/greptimedb-operator/api/v1alpha1
```
Comment (zyy17) [inline PROJECT:17]: Is it the right path? Maybe we should use `apis/`?

File: PROJECT [hunk 2]
```diff
@@ -10,14 +14,20 @@ resources:
...
+    validation: true
+    webhookVersion: v1
 - api:
     crdVersion: v1
     namespaced: true
   controller: true
   domain: greptime.io
   kind: GreptimeDBStandalone
-  path: github.com/greptime/greptimedb-operator/apis/v1alpha1
+  path: github.com/GreptimeTeam/greptimedb-operator/api/v1alpha1
```
Comment (zyy17) [inline PROJECT:28]: ditto

**Review by zyy17 — COMMENTED** [review]
mostly lgtm

File: PROJECT (hunk 1 above)
Comment (daviderli614) [reply PROJECT]: Now is `apis`.

**Review by zyy17 — APPROVED** [review]

**Pull request merged**